Webpack 2 + PostCSS errors (Invalid configuration object ...)

I have tested `webpack.config.js` exactly as in 
https://github.com/andywer/webpack-blocks#usage
which lead to errors:

```bash
± |master ✓| → npm run build

> webpack2-blocks-test@0.1.0 build /Users/dmitrizaitsev/Dropbox/Sandbox/webpack2-blocks-test
> webpack

Invalid configuration object. Webpack has been initialised using a configuration object that does not match the API schema.
 - configuration has an unknown property 'postcss'. These properties are valid:
   object { amd?, bail?, cache?, context?, dependencies?, devServer?, devtool?, entry, externals?, loader?, module?, name?, node?, output?, performance?, plugins?, profile?, recordsInputPath?, recordsOutputPath?, recordsPath?, resolve?, resolveLoader?, stats?, target?, watch?, watchOptions? }
   For typos: please correct them.
   For loader options: webpack 2 no longer allows custom properties in configuration.
     Loaders should be updated to allow passing options via loader options in module.rules.
     Until loaders are updated one can use the LoaderOptionsPlugin to pass these options to the loader:
     plugins: [
       new webpack.LoaderOptionsPlugin({
         // test: /\.xxx$/, // may apply this only for some modules
         options: {
           postcss: ...
         }
       })
     ]

npm ERR! Darwin 12.6.0
npm ERR! argv "/usr/local/bin/node" "/Users/dmitrizaitsev/.npm-global/bin/npm" "run" "build"
npm ERR! node v6.9.2
npm ERR! npm  v4.0.5
npm ERR! code ELIFECYCLE
npm ERR! webpack2-blocks-test@0.1.0 build: `webpack`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the webpack2-blocks-test@0.1.0 build script 'webpack'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the webpack2-blocks-test package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     webpack
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs webpack2-blocks-test
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls webpack2-blocks-test
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/dmitrizaitsev/Dropbox/Sandbox/webpack2-blocks-test/npm-debug.log

```

Here is the repo to reproduce the errors:

https://github.com/dmitriz/webpack-blocks-wp2-test

--- a/packages/postcss/CHANGELOG.md
+++ b/packages/postcss/CHANGELOG.md
@@ -1,5 +1,9 @@
 # @webpack-blocks/postcss - Changelog
 
+## 0.3.2
+
+- Bug fix: PostCSS plugin configuration now works with webpack 2 ([#68](https://github.com/andywer/webpack-blocks/issues/68))
+
 ## 0.3.1
 
 - Supporting custom PostCSS options now (`parser`, `stringifier`, `syntax`)

--- a/packages/postcss/README.md
+++ b/packages/postcss/README.md
@@ -20,6 +20,17 @@ module.exports = createConfig([
 ])
 ```
 
+Instead of passing the PostCSS plugins as an array you can also create a `postcss.config.js` file containing the plugin configuration (see [PostCSS loader usage](https://github.com/postcss/postcss-loader#usage)):
+
+```js
+// postcss.config.js
+module.exports = {
+  plugins: [
+    require('precss')
+  ]
+}
+```
+
 
 ## Options
 

--- a/packages/postcss/index.js
+++ b/packages/postcss/index.js
@@ -27,17 +27,36 @@ function postcss (plugins, options) {
     options.syntax && { syntax: options.syntax }
   )
 
-  return (context) => Object.assign({
-    module: {
-      loaders: [
-        {
-          test: context.fileType('text/css'),
-          exclude: Array.isArray(exclude) ? exclude : [ exclude ],
-          loaders: [ 'style-loader', 'css-loader', 'postcss-loader?' + JSON.stringify(postcssOptions) ]
-        }
+  return (context) => Object.assign(
+    {
+      module: {
+        loaders: [
+          {
+            test: context.fileType('text/css'),
+            exclude: Array.isArray(exclude) ? exclude : [ exclude ],
+            loaders: [ 'style-loader', 'css-loader', 'postcss-loader?' + JSON.stringify(postcssOptions) ]
+          }
+        ]
+      }
+    },
+    plugins ? createPostcssPluginsConfig(context.webpack, plugins) : {}
+  )
+}
+
+function createPostcssPluginsConfig (webpack, plugins) {
+  const isWebpack2 = typeof webpack.validateSchema !== 'undefined'
+
+  if (isWebpack2) {
+    return {
+      plugins: [
+        new webpack.LoaderOptionsPlugin({
+          options: { postcss: plugins }
+        })
       ]
     }
-  }, plugins ? {
-    postcss: plugins
-  } : {})
+  } else {
+    return {
+      postcss: plugins
+    }
+  }
 }

--- a/packages/postcss/package.json
+++ b/packages/postcss/package.json
@@ -1,6 +1,6 @@
 {
   "name": "@webpack-blocks/postcss",
-  "version": "0.3.1",
+  "version": "0.3.2",
   "description": "Webpack block for PostCSS.",
   "main": "lib/index",
   "license": "MIT",

--- a/packages/webpack2/__e2e-fixtures__/babel-postcss-extract-text/postcss.config.js
+++ b/packages/webpack2/__e2e-fixtures__/babel-postcss-extract-text/postcss.config.js
@@ -1,7 +1,0 @@
-const precss = require('precss')
-
-module.exports = {
-  plugins: [
-    precss
-  ]
-}

--- a/packages/webpack2/__e2e-fixtures__/babel-postcss-extract-text/webpack.config.js
+++ b/packages/webpack2/__e2e-fixtures__/babel-postcss-extract-text/webpack.config.js
@@ -12,6 +12,7 @@ const babel = require('@webpack-blocks/babel6')
 const postcss = require('@webpack-blocks/postcss')
 const extractText = require('@webpack-blocks/extract-text2')
 const path = require('path')
+const precss = require('precss')
 
 module.exports = createConfig([
   entryPoint(
@@ -21,7 +22,9 @@ module.exports = createConfig([
     path.join(__dirname, 'build/bundle.js')
   ),
   babel(),
-  postcss(),
+  postcss([
+    precss
+  ]),
   extractText(
     path.join('styles.css')
   ),

